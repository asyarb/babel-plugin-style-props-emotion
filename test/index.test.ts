import { PluginItem, transformSync } from '@babel/core'
import react from '@babel/preset-react'
import emotionPreset from '@emotion/babel-preset-css-prop'
import styleProps from 'babel-plugin-style-props'
import emotionAdapter from '../src'

const presets = [react, emotionPreset]
const plugins = [
  [styleProps, { stripProps: true }],
  [emotionAdapter, { stripProp: true }],
]

const parseCode = (example: string, plug?: PluginItem[]) =>
  transformSync(example, { plugins: plug || plugins, presets })!.code

describe('style props', () => {
  it('parses style props', () => {
    const result = parseCode(`
      <div color='tomato' bg='blue' />
    `)

    expect(result).toMatchSnapshot()
  })
})
