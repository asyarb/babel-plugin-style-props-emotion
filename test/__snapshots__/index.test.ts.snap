// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style props parses style props 1`] = `
"import _css from \\"@emotion/css\\";
import { jsx as ___EmotionJSX } from \\"@emotion/core\\";
import { getStyle as __getStyle, getScaleStyle as __getScaleStyle } from \\"babel-plugin-style-props-emotion/runtime\\";

___EmotionJSX(\\"div\\", {
  css:
  /*#__PURE__*/
  _css({
    color: __getStyle(theme, \\"colors\\", 'tomato'),
    backgroundColor: __getStyle(theme, \\"colors\\", 'blue')
  })
});"
`;
